pkg/controller/postgresqldatabase/postgresqldatabase_controller.go

```diff
@@ -0,0 +1,235 @@
+// and what is in the PostgreSQLDatabase.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcilePostgreSQLDatabase) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling PostgreSQLDatabase")
+
+	// Fetch the PostgreSQLDatabase instance
+	database := &lunarwayv1alpha1.PostgreSQLDatabase{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, database)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	reqLogger = reqLogger.WithValues("database", database.Spec.Name)
+	reqLogger.Info("Reconciling PostgreSQLDatabase")
+
+	// Resolve the password, is the value in a configMap or Secret or just a plain value
+	password, err := r.resolveDatabasePassword(*database, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Ensure the database is in sync with the object
+	err = r.ensurePostgreSQLDatabase(reqLogger, database.Spec.Name, password)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func postgresqlConnection(connectionString string) (*sql.DB, error) {
+	log.Info("Connecting to database", "url", connectionString)
+	db, err := sql.Open("postgres", connectionString)
+	if err != nil {
+		return nil, err
+	}
+	err = db.Ping()
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func (r *ReconcilePostgreSQLDatabase) resolveDatabasePassword(db lunarwayv1alpha1.PostgreSQLDatabase, namespaceName string) (string, error) {
+	if db.Spec.Password.Value != "" {
+		return db.Spec.Password.Value, nil
+	}
+
+	if db.Spec.Password.ValueFrom.SecretKeyRef.Key != "" {
+		return r.getSecretValue(db.Spec.Password.ValueFrom.SecretKeyRef.Name, namespaceName, db.Spec.Password.ValueFrom.SecretKeyRef.Key)
+	}
+
+	if db.Spec.Password.ValueFrom.ConfigMapKeyRef.Key != "" {
+		return r.getConfigMapValue(db.Spec.Password.ValueFrom.ConfigMapKeyRef.Name, namespaceName, db.Spec.Password.ValueFrom.ConfigMapKeyRef.Key)
+	}
+
+	return "", fmt.Errorf("no password")
+}
+
+func (r *ReconcilePostgreSQLDatabase) ensurePostgreSQLDatabase(log logr.Logger, name, password string) error {
+	log.Info(fmt.Sprintf("Creating service user with name %s and password %s", name, password))
```

Comment:
I guess this is OK for now, but we should be very sure to avoid printing the password before using this for anything real.

Reply:
It's not necessary anymore. I will just delete it